x/subscription/README.md

```diff
@@ -0,0 +1,180 @@
+After choosing a plan, a consumer can perform the purchase using the `buy` transaction command:
+
+```bash
+lavad tx subscription buy [plan-index] [optional: consumer] [optional: duration(months)] [flags]
+```
+
+Once a consumer bought a subscription, a new instance of Subscription object will be saved into the state:
+
+```go
+struct Subscription {
+	Creator            string              // creator pays for the subscription
+	Consumer           string              // consumer uses the subscription
+	Block              uint64              // when the subscription was last recharged
+	PlanIndex          string              // index (name) of plan
+	PlanBlock          uint64              // when the plan was created
+	DurationBought     uint64              // total requested duration in months
+	DurationLeft       uint64              // remaining duration in months
+	MonthExpiryTime    uint64              // expiry time of current month
+	MonthCuTotal       uint64              // CU allowance during current month
+	MonthCuLeft        uint64              // CU remaining during current month
+	Cluster            string              // cluster key
+	DurationTotal      uint64              // continuous subscription usage in months
+	AutoRenewal        bool                // automatic renewal when the subscription expires
+	FutureSubscription *FutureSubscription // future subscription made with buy --advance-purchase
+}
+
+struct FutureSubscription {
+	Creator        string // creator pays for the future subscription. Will replace the original one once activated
+	PlanIndex      string // index (name) of plan
+	PlanBlock      uint64 // when the plan was created
+	DurationBought uint64 // total requested duration in months
+}
+```
+
+When a consumer buys a subscription, an admin project is automatically generated for them.  
+When the creator and consumer are different individuals, indicating that someone is purchasing a subscription for another user, the tokens will be taken out of the creator's account.
+
+Subscriptions are saved in a fixation store (see [FixationStore](https://github.com/lavanet/lava/blob/main/x/fixationstore/README.md) module for reference).
+
+There could only be a single active subscription for a consumer.  
+When a consumer purchase a subscription, the tokens are transferred from their account immediately.  
+Those tokens are later on distributed among the providers that served that consumer, and the delegators that staked those providers.
+
+### Advance Month
+
+One of the primary processes within this module involves the function `advanceMonth`, which is invoked upon the expiration of a timer set by the subscription. This function holds significance as it manages the logic that occurs at the conclusion of each month of a subscription.
+
+The advanceMonth function plays a crucial role in the subscription lifecycle, performing several important tasks:
```

Comment:
add the task of paying all providers that served this subscription

Reply:
this is also described in pairing

Reply:
Added in 1898ee2ce8245519bcd5d0d2b24d8bba5d5e04f0